src/main/java/io/codelex/flightplanner/repository/FlightPlannerRepository.java

```diff
@@ -0,0 +1,122 @@
+package io.codelex.flightplanner.repository;
+
+import io.codelex.flightplanner.Airport;
+import io.codelex.flightplanner.Flight;
+import io.codelex.flightplanner.PageResult;
+import io.codelex.flightplanner.SearchFlightRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Repository
+public class FlightPlannerRepository {
+    private final List<Flight> flights = new ArrayList<>();
+    private int flightId;
+
+    public void clearFlights() {
+        flights.clear();
+    }
+
+    public synchronized Flight addFlight(@RequestBody Flight flight) {
+        if (flights.contains(flight)) {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        }
+
+        if (isFromAndToSameAirport(flight)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        if (hasInvalidDates(flight)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+        }
+
+        flight.setId(flightId);
+        flightId++;
+        flights.add(flight);
+        return flights.get(flights.size() - 1);
+    }
+
+    public Flight fetchFlight(int id) {
+        if (isExistingFlight(id)) {
+            return findExistingFlight(id);
+        }
+        throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+    }
+
+    public synchronized void deleteFlight(int id) {
+        if (isExistingFlight(id)) {
+            Flight foundFlight = findExistingFlight(id);
+            flights.remove(foundFlight);
+        }
+    }
+
+    public List<Airport> searchAirports(String search) {
+        List<Airport> airportList = getExistingAirportList(search);
+        if (airportList.isEmpty()) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+        }
+        return airportList;
+    }
+
+    public PageResult searchFlights(SearchFlightRequest flight) {
+        PageResult pageResult = getExistingFlight(flight);
+        if (flight.getTo().equals(flight.getFrom())) {
```

Comment:
Es ļoti ieteiktu pārsaukt to mainīgo flight uz kaut ko citu. :D Jo faktiski tas nav flight. Droši vari saukt arī vienkārši par request.